android/src/main/java/com/tanguyantoine/react/MusicControlModule.java

```diff
@@ -205,6 +211,16 @@ synchronized public void setNowPlaying(ReadableMap metadata) {
         nb.setContentInfo(album);
         nb.setColor(notificationColor);
 
+        hasCustomIcon = notificationIcon != null ? true : false;
+        if (hasCustomIcon) {
+          int smallIcon = loadIconResource(notificationIcon);
+          if (smallIcon != 0) {
+            nb.setSmallIcon(smallIcon);
+          } else {
+            hasCustomIcon = false;
+          }
+        }
+
```

Comment:
Can't you bring back `setCustomNotificationIcon` from the notification class and just call it from here? It would reduce a lot of code. Also, there is no reason for `hasCustomIcon`

Reply:
so basically, the setCustomNotificationIcon was replacing the value of `smallIcon` class variable in the Notification. If it was used with the `setNowPlaying`, it would mean that if the developer wanted to change the icon for a specific nowPlaying event, but then didn't specify it on the next call of `setNowPlaying` they might expect a default icon they had, but now that would be overwritten by the icon they specified in the previous call. Doing it this way, allows us to set the notification on the individual call to `setNowPlaying` if specified as an option, and fall back to default if not specified.

The `hasCustomIcon` was then added to prevent any calls from show to run `builder.setSmallIcon` inside of the notification code, as that would overwrite the custom icon. It could have been used as a local variable in the `setNowPlaying` and just passed down, but since `updateNowPlaying` also makes a conditional call to `show` that could cause the custom icon to be overwritten.

Reply:
If no custom icon is specified, you can load the `play` icon back

Reply:
Basically, here's my train of thought, and I might be wrong.

Developer has an icon `music_control_icon` that they use as the default one for their notifications. When they want to use a custom icon, they specify `notificationIcon` to something like `my_custom_icon` in the `setNowPlaying` in some specific case. Then they call `setNowPlaying` again without the the `notificationIcon`, they expect the notification icon to be `music_control_icon`

In the original code I wrote, the `setCustomNotificationIcon` would set the value of `smallIcon` variable of the `MusicControlNotification`. If `setNowPlaying` called `setCustomNotificationIcon` when a `notificationIcon` was set, it would basically overwrite the `smallIcon` variable value in the instance of the notification, which will result in the icon specified once being used for any future notifications, even if `notificationIcon` was not specified.

The version of code I pushed in the latest update of this PR does the following. `smallIcon` from the `MusicControlNotification` never gets touched, so after the instance is created, the value of `smallIcon` will be a reference either to `play` or to `music_control_icon` if it was provided. Instead what it does, is if the developer passed down a `notificationIcon` to the `setNowPlaying` call, it would try to load that resource, and if the resource is available it would set it on the NotificationBuilder instance being passed down to the `show` call. The obvious result of this is if the `notificationIcon` is not provided, the NotificationBuilder being passed down to the show call would not have an icon set on it, and the notification uses whatever the value of `smallIcon` was set to when the Notification instance was originally created.

Which then brings us back to the `hasCustomIcon` boolean flag, and the need for it. In the `MusicControlNotification` the `show` method will call `builder.setSmallIcon(smallIcon)`. Because the change in `setNowPlaying` and it being a safer way to not touch the value of `smallIcon`, any call to the `show` function would result in the icon being set to whatever the value of `smallIcon` is, thus overriding the change from the `setNowPlaying` call. However if we can pass something like `hasCustomIcon` we can basically prevent that from happening.

Maybe I'm missing something, but regarding the code reduction, I don't see how it would be more or less code, or in any way affect the complexity? It's still essentially the same code.

Reply:
The way I would do it is:
* Create a `customIcon` field in `MusicControlNotification`
* Bring back `setCustomNotificationIcon`, but setting `customIcon` instead of `smallIcon`
* Accept null values in `setCustomNotificationIcon`, which would set `customIcon` to -1
* Check whether `customIcon` is -1 in `MusicControlNotification.show`, if it isn't, then use it instead of `smallIcon`

There is also an even simpler way to achieve this:
* Bring back `setCustomNotificationIcon`, but using `builder.setSmallIcon` directly
* Accept null values in `setCustomNotificationIcon`, which would `builder.setSmallIcon` with the `smallIcon` field
* Remove calls to `builder.setSmallIcon` in `MusicControlNotification.show`

That way, it would be much cleaner. Is there any other reason to use your code that I'm not seeing? Thanks :)

Reply:
ok, cool, wasn't sure we wanted to rip out `setSmallIcon` from `show`. With that in mind, yeah let me just push an update, that should make this simpler. Thanks again